stdlib/public/core/BidirectionalCollection.swift

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// TODO: swift-3-indexing-model - Add in BidirectionalCollection protocol documentation
+public protocol BidirectionalCollection : Collection {
+  // TODO: swift-3-indexing-model - replaces functionality in BidirectionalIndex
+  /// Returns the previous consecutive `Index` in a discrete sequence of
+  /// `Index` values.
+  ///
+  /// If `i` has a well-defined successor, `self.previous(self.next(i)) == i`.
+  /// If `i` has a well-defined predecessor, `self.next(self.previous(i)) == i`.
+  ///
+  /// - Precondition: `i` has a well-defined predecessor.
+  @warn_unused_result
+  func previous(i: Index) -> Index
+  
+  // TODO: swift-3-indexing-model - replaces functionality in BidirectionalIndex
+  func _previousInPlace(inout i: Index)
+}
+
+/// Default implementation for bidirectional indexable collections
```

Comment:
All collections are "indexable", so mentioning that is redundant.  We also put periods at the end of comments, so signal that the comment, even if it is a sentence fragment, was typed completely as the author expected.